components/modal/frame.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * External dependencies
+ */
+import clickOutside from 'react-click-outside';
+import { forEach } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, compose, createRef } from '@wordpress/element';
+import { keycodes } from '@wordpress/utils';
+import { focus } from '@wordpress/dom';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import withFocusReturn from '../higher-order/with-focus-return';
+import withFocusContain from '../higher-order/with-focus-contain';
+import withGlobalEvents from '../higher-order/with-global-events';
+import withSafeTimeout from '../higher-order/with-safe-timeout';
+
+const {
+	ESCAPE,
+} = keycodes;
+
+class ModalFrame extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.containerRef = createRef();
+		this.handleKeyDown = this.handleKeyDown.bind( this );
+		this.handleClickOutside = this.handleClickOutside.bind( this );
+		this.focusFirstFocusable = this.focusFirstFocusable.bind( this );
+		this.focusFirstTabbable = this.focusFirstTabbable.bind( this );
+	}
+
+	/**
+	 * Focuses the first tabbable element when props.focusOnMount is true.
+	 */
+	componentDidMount() {
+		// Focus on mount
+		if ( this.props.focusOnMount ) {
+			const { setTimeout } = this.props;
+			// Required because the node is appended to the DOM after rendering.
+			setTimeout( () => {
```

Comment:
`setTimeout` is a code smell and should be avoided. See https://github.com/WordPress/gutenberg/issues/7461#issuecomment-399990425 as an example for the type of breakage we risk here. We need to find an alternative solution which does not require delayed execution.